container_files/demos/Virtual manipulation of datasets.ipynb

```diff
@@ -0,0 +1,1583 @@
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "collapsed": true
+   },
+   "outputs": [],
+   "source": [
+    "from pymldb import Connection\n",
+    "mldb = Connection(\"http://localhost\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Loading the Dataset\n",
+    "A powerful feature of MLDB allows us to execute SQL code while loading data. Here, the [`tokenize`](../../../../doc/#builtin/sql/ValueExpression.md.html#builtinfunctions) function first splits the columns into (lowercased) individual words, removing words with a length of less than 3 characters. Notice that we have chosen to use a [sparse mutable matrix](../../../../doc/#builtin/datasets/MutableSparseMatrixDataset.md.html) since we do know ahead time the column size for each row."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 13,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "<Response [201]>\n"
+     ]
+    }
+   ],
+   "source": [
+    "print mldb.put('/v1/procedures/import_ML_concepts', {\n",
+    "    \"type\":\"import.text\",\n",
+    "    \"params\": {\n",
+    "        \"dataFileUrl\":\"file:///mldb_data/datasets/MachineLearningConcepts.csv\",\n",
+    "        \"outputDataset\": {\n",
+    "                            \"id\":\"ml_concepts\", # can be accessed later using id\n",
+    "                            \"type\": \"sparse.mutable\" # sparse mutable dataset is needed since we tokenize words below\n",
+    "                         },\n",
+    "        \"named\": \"Concepts\", #Row name expression for output dataset\n",
+    "        \"select\": \"\"\"     tokenize(\n",
+    "                                lower(Text), \n",
+    "                                {splitchars: ' -''\"?!;:/[]*,().',  \n",
+    "                                min_token_length: 3}) as *\n",
+    "                                \n",
+    "                         \"\"\", # Within tokenize function:\n",
+    "                                # lower case\n",
+    "                                # leave out punctuation and spaces\n",
+    "                                # allow only words > 3 characters\n",
+    "        \"runOnCreation\": True\n",
+    "    }\n",
+    "})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## A quick look at the data\n",
+    "\n",
+    "We can use the [Query API](../../../../doc/#builtin/sql/QueryAPI.md.html) to get the data into a Pandas DataFrame to take a quick look at it. You will notice that certain cells have a 'NaN' value as seen below. This occurs when the word is not found in the description of the concept. MLDB does uses 'NaN' to more efficiently reprensent a sparse matrix (TBD with François)."
```

Comment:
En fait c'est pandas qui fait un NaN. "You will notice that certain cells have a value of 'NaN'. This is because the dataset is sparse: every word is not present in the description of every concept. Those missing values are representend as NaNs in a Pandas DataFrame."